src/allotropy/parsers/thermo_fisher_genesys30/thermo_fisher_genesys30_structure.py

```diff
@@ -0,0 +1,147 @@
+""" Structure file for ThermoFisher Genesys 30 Adapter """
+from __future__ import annotations
+
+from datetime import datetime
+import io
+
+import pandas as pd
+
+from allotropy.allotrope.models.adm.spectrophotometry.benchling._2023._12.spectrophotometry import (
+    ContainerType,
+)
+from allotropy.allotrope.models.shared.definitions.definitions import (
+    FieldComponentDatatype,
+)
+from allotropy.allotrope.schema_mappers.adm.spectrophotometry.benchling._2023._12.spectrophotometry import (
+    Data,
+    DataCube,
+    DataCubeComponent,
+    Measurement,
+    MeasurementGroup,
+    MeasurementType,
+    Metadata,
+)
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.constants import NOT_APPLICABLE
+from allotropy.parsers.lines_reader import LinesReader, read_to_lines
+from allotropy.parsers.thermo_fisher_genesys30 import constants
+from allotropy.parsers.utils.pandas import df_to_series_data
+from allotropy.parsers.utils.uuids import random_uuid_str
+from allotropy.parsers.utils.values import try_float_or_none
+
+
+def create_data(named_file_contents: NamedFileContents) -> Data:
+    """
+    Structures the named_file_contents according to the mapper file
+    :param named_file_contents: The contents of the file to convert
+    :return: The data according to spectrophotometer data mapper
+    """
+    filename = named_file_contents.original_file_name
+
+    lines = read_to_lines(named_file_contents)
+    reader = LinesReader(lines)
+
+    if filename.endswith(".csv"):
+        metadata_list = [line for line in reader.pop_until("^,,") if line]
+        reader.drop_until_inclusive(",,")
+        delimiter = ","
+    elif filename.endswith(".tsv"):
+        metadata_list = list(reader.pop_until_empty())
+        reader.drop_until_inclusive("")
+        delimiter = "\t"
+    else:
+        message = f"{constants.UNSUPPORTED_FILE_FORMAT_ERROR} {filename}"
+        raise AllotropeConversionError(message)
+
+    rawdata_list = list(reader.pop_until_empty())
+    rawdata_string = io.StringIO("\n".join(rawdata_list))
+    rawdata_dataframe = pd.read_csv(rawdata_string, header=0, delimiter=delimiter)
+
+    metadata_string = io.StringIO("\n".join(metadata_list))
+    metadata_dataframe = pd.read_csv(
+        metadata_string,
+        header=None,
+        delimiter=delimiter,
+        keep_default_na=False,
+        index_col=0,
+    ).T
+    metadata_dataframe.columns = metadata_dataframe.columns.str.strip()
+    rawdata_dataframe.columns = rawdata_dataframe.columns.str.strip()
+
+    metadata_dataframe = metadata_dataframe.head(1)
+
+    metadata = df_to_series_data(metadata_dataframe, "Failed to parser header data")
+
+    experiment_name = metadata.get(str, "Scan")
+
+    experiment_type, measurement_time = get_date_time(experiment_name)
```

Comment:
Very nitty - but can you rename this to show that you are getting experiment type and time, e.g. `get_measurment_type_and_time`

Reply:
I think renaming the function to 'get_experiment_type_and_time' might make it clearer and more accurate, as the function retrieves the experiment type rather than just the measurement type. Could you share your thought process behind preferring 'measurement type' over 'experiment type' in this context?

Reply:
`get_experiment_type_and_time` is fine as well, just didn't want it to be `get_date_time` when it returns 2 values

Reply:
okay Nates, thanks for the feedback! I'll stick with `get_experiment_type_and_time` to keep it clear that it returns two values.